### ISSUE TITLE
 NSLog system log "spam" in iOS commonAPI code

### ISSUE THREAD

**Issue by jtara** [rhomobile/rhodes#643]
![suchnslog](https://cloud.githubusercontent.com/assets/958610/8017294/ca8bc9d8-0ba6-11e5-9f1d-77a18189acf6.jpg)

Many, many, many new commonAPI modules are using `NSLog()` in iOS Objective-C for logging. There is no way to control these messages, as they are not being routed through the normal Rhodes logging system.

`NSLog()` lacks any mechanism for filtering, "levels" etc. 

`asl`, or "Apple System Log" is a more proper facility to use, as it supports levels.

Rhodes has a set of `RAWLOG` macros for iOS (`rholog.h`) that had been used consistently prior to new commonAPI modules. These route messages through the normal Rhodes logging facility.

`RAWLOG` macros are a pain to use in Objective-C code, because they only support C strings. (I think...) Yes, I've been tempted to use `NSLog()` as well in my own native extensions. But it's just plain wrong, and leaks all sorts of stuff in the system log that ought not to be leaked!

See:

http://doing-it-wrong.mikeweller.com/2012/07/youre-doing-it-wrong-1-nslogdebug-ios.html

Rhodes is in good company in logging lines to the system log in production that are not errors. But that's no excuse. One day, Apple is likely to crack-down!

Anything more verbose than error (e.g. info, trace) should not go to system log except by explicit option set by user or developer for debugging purposes. That's already the case in Rhodes if logging goes through the normal log system. But not with direct use of NSLog. There isn't even a way to designate the logging level.

**Comment by jtara** [comment]
I notice in latest repo even more NSLog tracing has been added.

IMO, tracing should be done using normal Rhodes logging, so that it can be controlled. Otherwise, these traces "leak" to the system log, which often is not desirable for production applications. It is a potential security risk, and might violate App Store policy, which wants system logging to be only about warnings and errors - not trace.

Did I miss some added ability to control these traces?

In the past, I have had to maintain a custom Rhodes with this tracing patched-out.
